### PR TITLE
async contracts support

### PR DESCRIPTION
**Pull request by haimbj1**

### CODE REVIEW

File: .github/workflows/releaseAsync.yml
Comment (liorbond) [inline .github/workflows/releaseAsync.yml:1]: release cofhe - async will be for async fhenix

File: async-contracts/AsyncFHE.sol
Comment (liorbond) [inline async-contracts/AsyncFHE.sol:1]: Please name the folder "cofhe-contracts" and the file FHE.sol

File: async-contracts/ICofhe.sol [hunk 1]
```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.25 <0.9.0;
+
+struct CiphertextKey {
+    bool     isTriviallyEncrypted;
+    uint8   uintType;
+    int32    securityZone;
+    uint256  hash;
+}
```
Comment (liorbond) [inline async-contracts/ICofhe.sol:9]: I know it doesn't related to ur pr but please remove this from here and move this to TaskManager.sol

File: async-contracts/ICofhe.sol [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    // Must match values defined in warp-drive protobufs for everything to
+    // make sense
+    uint8 internal constant EUINT8_TFHE = 0;
+    uint8 internal constant EUINT16_TFHE = 1;
+    uint8 internal constant EUINT32_TFHE = 2;
+    uint8 internal constant EUINT64_TFHE = 3;
+    uint8 internal constant EUINT128_TFHE = 4;
+    uint8 internal constant EUINT256_TFHE = 5;
+    uint8 internal constant EADDRESS_TFHE = 12;
+    // uint8 internal constant INT_BGV = 12;
```
Comment (liorbond) [inline async-contracts/ICofhe.sol:70]: Remove please

File: async-contracts/package.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+	"name": "@fhenixprotocol/async-contracts",
+	"description": "Smart Contract Library for the Async Fhenix CoFHE with FHE primitives",
```
Comment (liorbond) [inline async-contracts/package.json:3]:
```suggestion
	"description": "Smart Contract Library for Fhenix CoFHE with FHE primitives",
```

File: async-contracts/ICofhe.sol (hunk 1 above)
Comment (haimbj1) [reply async-contracts/ICofhe.sol:9]: https://github.com/FhenixProtocol/copro-task-manager-hardhat/pull/14

**Review by liorbond — APPROVED** [review]

**Pull request merged**